### PR TITLE
fix CID 1441972

### PR DESCRIPTION
**Pull request by wsnipex**
return a string instead of a dangling pointer

## Description

The only consumer of this function is std::string anyway, so there is no really good reason to stick to char*

## Motivation and Context

```
*** CID 1441972:  Memory - illegal accesses  (WRAPPER_ESCAPE)
/build/build/xbmc/CompileInfo.cpp: 69 in CCompileInfo::GetBuildDate()()
63     const char* CCompileInfo::GetBuildDate()
64     {
65       const std::string bdate = "20181225";
66       if (!bdate.empty())
67       {
68         std::string datestamp = bdate.substr(0, 4) + "-" + bdate.substr(4, 2) + "-" + bdate.substr(6, 2);
>>>     CID 1441972:  Memory - illegal accesses  (WRAPPER_ESCAPE)
>>>     The internal representation of local "datestamp" escapes, but is destroyed when it exits scope.
69         return datestamp.c_str();
70       }
71       return "1970-01-01";
```

## How Has This Been Tested?

built on ubuntu

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/CompileInfo.cpp.in [hunk 1]
```diff
@@ -60,13 +60,13 @@ const char* CCompileInfo::GetCopyrightYears()
   return "@APP_COPYRIGHT_YEARS@";
 }
 
-const char* CCompileInfo::GetBuildDate()
+const std::string CCompileInfo::GetBuildDate()
```
Comment (Rechi) [inline xbmc/CompileInfo.cpp.in]: Either pass by const reference or by non-const value.
I don't see any benefit for passing by const value.

Comment (wsnipex) [reply xbmc/CompileInfo.cpp.in]: update to pass by non-const value

**Review by yol — APPROVED** [review]
All good. Maybe we could just change everything to `std::string` for consistency

**Comment by Rechi** [comment]
jenkins build this please

**Review by Rechi — APPROVED** [review]

**Pull request merged**